src/DI/ContainerBuilder.php

```diff
@@ -416,6 +416,8 @@ private function resolveImplement(ServiceDefinition $def, $name)
 						throw new ServiceCreationException("Type hint for \${$param->getName()} in $interface::$methodName() doesn't match type hint in $class constructor.");
 					}
 					$def->getFactory()->arguments[$arg->getPosition()] = self::literal('$' . $arg->getName());
+				} elseif (!$def->getSetup()) {
```

Comment:
When implementing generated factory there are only two ways to use parameters in `create` method:
a) in constructor of created object.
b) in setup directives.
Or am I wrong?